src/wh_server.c

```diff
@@ -294,7 +294,13 @@ int wh_Server_HandleRequestMessage(whServerContext* server)
 
         /* Send a response */
         /* TODO: Respond with ErrorResponse if handler returns an error */
-        if (rc == 0) {
+        if (rc == 0 || rc == WH_ERROR_CANCEL) {
+            /* notify the client that their request was canceled */
+            if (rc == WH_ERROR_CANCEL) {
+                kind = WH_MESSAGE_KIND(WH_MESSAGE_GROUP_CANCEL, 0);
```

Comment:
Should we make actions for the cancel message group? Even if there is just one for now, probably a good idea